### DISCUSSION TITLE
How to access database tables in Edge Functions

### DISCUSSION THREAD

**Discussion by ijaspreetsinghh** [supabase/supabase#6321] [Questions] [answered]
Supabase has newly introduced [Edge Functions](https://supabase.com/blog/2022/03/31/supabase-edge-functions)  a Function-as-a-Service . This seems to be very cool as any function can be called with just an API click without worrying about extra servers, but documentation to connect the function to a database and access tables are not provided.  I'm looking for a way to access data tables and make CRUD operations to data using edge functions.

**Comment by GaryAustin1** [comment, marked as answer]
You use the Supabase-js client.  SB keys are provided to function. 
https://github.com/supabase/supabase/blob/master/examples/edge-functions/supabase/functions/select-from-table-with-auth-rls/index.ts

**Reply by ijaspreetsinghh** [reply]
That helped a lot, thanks man 👍

**Reply by MarkBennett** [reply]
Does anyone have any benchmarks for supabase db queries? Are any benefits from function edge deploys offset by the round trip times to access a central db region?

**Reply by GaryAustin1** [reply]
@MarkBennett you should start a new discussion.  I've not seen anything specifically bench marking your concern.

**Reply by MarkBennett** [reply]
Will do @GaryAustin1. Thanks! 😄

**Reply by MarkBennett** [reply]
Done!

https://github.com/supabase/supabase/discussions/8598

**Reply by qrhfz** [reply]
@GaryAustin1 link is dead

**Reply by corbinpage** [reply]
Updated link: https://github.com/supabase/examples/blob/main/supabase-js-v1/edge-functions/supabase/functions/select-from-table-with-auth-rls/index.ts

**Reply by guillaumeduhan** [reply]
Hello guys and thanks for your answers. This part is clearly not well explained! We need to work on documentation :)

**Reply by twgonzalez** [reply]
If this was in the documentation, it would have saved me a day or more of time stuggling through having edge functions connect both locally and hosted.
